### PR TITLE
add proposal/7.5.1 and update recipe for conan2

### PR DESCRIPTION
**Pull request by Jean1995**
Specify library name and version:  **proposal/7.5.1**

This recipe add version 7.5.1, and I have tried my best to modernize the package to be compatible with conan2 (while still remaining compatible with conan 1.x for now).

I'm open for feedback 🙂 We have users that would like to use the newest version of proposal, so I would be very glad if someone could find the time to review my PR.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Jean1995** [comment]
The conan v2 pipeline fails because the cubicinterpolation package has not yet been updated for conan v2. I have already created a PR for this (#16299), but unfortunately this hasn't been reviewed yet.

File: recipes/proposal/all/test_package/conanfile.py [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
-
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake, cmake_layout
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "CMakeToolchain", "CMakeDeps", "VirtualRunEnv"
+    requires = "spdlog/1.9.2" # TODO: not sure why this is explicitly necessary, but it fails without it
```
Comment (prince-chrismc) [inline recipes/proposal/all/test_package/conanfile.py:9]: Conan 2.0 changes the graph model and deps are private by default to encourage good software practices.

you need to add a `transitive_header=True` with a link to the reason why it's need (ie a comment from the bot failing is perfect or a link to the source code)

**Review by prince-chrismc — COMMENTED** [review]
Same comments are your other PR, pretty easy ones

you need to change https://github.com/conan-io/conan-center-index/pull/16851/files#diff-f43c124420645bac1f2bb5eaa326a41c40865fc54f9f2b43420d3100b21335ffR49 to use `rm_safe` too pretty please

File: recipes/proposal/all/test_package/conanfile.py (hunk 1 above)
Comment (Jean1995) [reply recipes/proposal/all/test_package/conanfile.py:9]: I am honestly note sure why I would need to put `transitive_headers=True` for spdlog, but not for any other libraries. 
If I don't (and do not explicitly write `requires = "spdlog/1.9.2"` in the conanfile of the test_package), I get the error
```
/usr/local/include/spdlog/fmt/fmt.h:27:14: fatal error: fmt/core.h: No such file or directory
   27 | #    include <fmt/core.h>
```
which sounds related to the experience in #11981 and #15832...

I could do this, and this does fix the issue, but I would like to understand why this is necessary 😅

**Comment by Jean1995** [comment]
I've been trying to reproduce the issue with the conan v2 pipeline locally, but was unsuccessful so far...
The error is
```
fatal error: nlohmann/json_fwd.hpp: No such file or directory
```

and if it is actually a problem, it might be related to #5188.

**Review by jcar87 — APPROVED** [review]
Thank you for updating this recipe to support Conan 2.0, and for re-ordering the versions in the yaml file! :)

**Review by prince-chrismc — APPROVED** [review]

File: recipes/proposal/all/conanfile.py [hunk 2]
```diff
@@ -57,48 +62,45 @@ def _minimum_compilers_version(self):
         return {"Visual Studio": "15", "gcc": "5", "clang": "5", "apple-clang": "5"}
```
Comment (SpaceIm) [inline recipes/proposal/all/conanfile.py:62]: you can add `"msvc": "191"` to this dictionary

File: recipes/proposal/all/conanfile.py [hunk 3]
```diff
@@ -26,29 +32,28 @@ class PROPOSALConan(ConanFile):
         "with_python": False,
     }
 
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
+    generators = "CMakeDeps"
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
```
Comment (SpaceIm) [inline recipes/proposal/all/conanfile.py:39]: a layout is missing, and this attribute could be removed

File: recipes/proposal/all/conanfile.py [hunk 4]
```diff
@@ -1,15 +1,21 @@
...
 
-required_conan_version = ">=1.43.0"
+required_conan_version = ">=1.53.0"
 
 
 class PROPOSALConan(ConanFile):
     name = "proposal"
     homepage = "https://github.com/tudo-astroparticlephysics/PROPOSAL"
     license = "LGPL-3.0"
     exports_sources = "CMakeLists.txt"
```
Comment (SpaceIm) [inline recipes/proposal/all/conanfile.py:17]: this file is not required anymore

**Comment by SpaceIm** [comment]
I've opened https://github.com/conan-io/conan-center-index/pull/17031

**Pull request merged**